Library/Homebrew/test/dev-cmd/bottle_spec.rb

```diff
@@ -270,15 +384,183 @@ class Testball < Formula
           url "file://#{tarball}"
           sha256 "#{tarball.sha256}"
 
+          option "with-foo", "Build with foo"
+
           bottle do
             root_url "#{HOMEBREW_BOTTLE_DEFAULT_DOMAIN}"
             cellar :any_skip_relocation
             sha256 "a0af7dcbb5c83f6f3f7ecd507c2d352c1a018f894d51ad241ce8492fa598010f" => :big_sur
             sha256 "5334dd344986e46b2aa4f0471cac7b0914bd7de7cb890a34415771788d03f2ac" => :catalina
           end
 
+          def install
+            (prefix/"foo"/"test").write("test") if build.with? "foo"
+            prefix.install Dir["*"]
+            (buildpath/"test.c").write \
+            "#include <stdio.h>\\nint main(){printf(\\"test\\");return 0;}"
+            bin.mkpath
+            system ENV.cc, "test.c", "-o", bin/"test"
+          end
+
+
+
+          # something here
+
+        end
+      EOS
+    )
+  end
+
+  it "adds the bottle block to a formula that has no bottle block using keep old" do
+    core_tap = CoreTap.new
+    core_tap.path.cd do
+      system "git", "init"
+      setup_test_formula("testball")
+      system "git", "add", "--all"
+      system "git", "commit", "-m", "testball 0.1"
+    end
+
+    begin
+      Pathname("testball-1.0.big_sur.bottle.json").write(
+        stub_hash(
+          {
+            "name":           "testball",
+            "version":        "1.0",
+            "path":           "#{core_tap.path}/Formula/testball.rb",
+            "cellar":         "any_skip_relocation",
+            "os":             "big_sur",
+            "filename":       "hello-1.0.big_sur.bottle.tar.gz",
+            "local_filename": "hello--1.0.big_sur.bottle.tar.gz",
+            "sha256":         "a0af7dcbb5c83f6f3f7ecd507c2d352c1a018f894d51ad241ce8492fa598010f",
+          },
+        ),
+      )
+
+      Pathname("testball-1.0.catalina.bottle.json").write(
+        stub_hash(
+          {
+            "name":           "testball",
+            "version":        "1.0",
+            "path":           "#{core_tap.path}/Formula/testball.rb",
+            "cellar":         "any_skip_relocation",
+            "os":             "catalina",
+            "filename":       "testball-1.0.catalina.bottle.tar.gz",
+            "local_filename": "testball--1.0.catalina.bottle.tar.gz",
+            "sha256":         "5334dd344986e46b2aa4f0471cac7b0914bd7de7cb890a34415771788d03f2ac",
+          },
+        ),
+      )
+
+      expect {
+        brew "bottle",
+             "--merge",
+             "--write",
+             "--keep-old",
+             "testball-1.0.big_sur.bottle.json",
+             "testball-1.0.catalina.bottle.json"
+      }.to output("Error: --keep-old was passed but there was no existing bottle block!\n").to_stderr
+    ensure
+      FileUtils.rm_f "testball-1.0.catalina.bottle.json"
+      FileUtils.rm_f "testball-1.0.big_sur.bottle.json"
+    end
+  end
+
+  it "adds the bottle block to a formula that has already a bottle block using keep old" do
```

Comment:
Also, it might be possible to surround this test and the other `--keep-old` test in a `context` block and delete the `when using --keep-old` part. For example:

```rb
context "with --keep-old" do
  it "updates the block ..." do
  ...
end
```

Reply:
I did not use the context to let the tests look similar. The first cases do not have a context so let's keep it simple.